src/node-file-trace.ts

```diff
@@ -153,30 +160,29 @@ export class Job {
     this.esmFileList = new Set();
     this.processed = new Set();
     this.warnings = new Set();
+
+    // This creates the cached versions of these IO functions
+    // The internal methods are only there in case the functions are overwritten from config and the overwritten function wants
+    // to call the internal implementation
+    this.stat = this.internalStat = cacheFileIOFactory(this.statFn.bind(this), this.statCache, this.fileIOQueue);
+    this.readFile = this.internalReadFile = cacheFileIOFactory(this.readFileFn.bind(this), this.fileCache, this.fileIOQueue);
+    this.readlink = this.internalReadlink = cacheFileIOFactory(this.readlinkFn.bind(this), this.symlinkCache, this.fileIOQueue);
```

Comment:
internal might actually not be a good choice for the name, because it makes it look like something purely internal one shouldn't use. Maybe rather `base` or `real`. Naming :D

Maybe you think it is not even necessary to have these methods, then I can also remove them and we need to figure out a different way in the tests.

Reply:
>  it makes it look like something purely internal one shouldn't use

I think it is internal and shouldn't be used. It was only added in PR #40 as a mock.